utilities_test.go

```diff
@@ -96,7 +97,88 @@ func TestDatasetWarp(t *testing.T) {
 		t.Errorf("wrong block size %d,%d", st.BlockSizeX, st.BlockSizeY)
 	}
 }
+func TestDatasetWarpMulti(t *testing.T) {
+	ds1, _ := Create(Memory, "", 3, Byte, 5, 5)
+	ds2, _ := Create(Memory, "", 3, Byte, 5, 5)
 
+	sr, _ := NewSpatialRefFromEPSG(4326)
+	_ = ds1.SetSpatialRef(sr)
+	_ = ds2.SetSpatialRef(sr)
+
+	_ = ds1.SetGeoTransform([6]float64{45, 1, 0, 35, 0, -1})
+	_ = ds2.SetGeoTransform([6]float64{50, 1, 0, 35, 0, -1})
+
+	for _, b := range ds1.Bands() {
+		_ = b.Fill(200, 0)
+	}
+
+	for _, b := range ds2.Bands() {
+		_ = b.Fill(100, 0)
+	}
+
+	defer ds1.Close()
+	defer ds2.Close()
+
+	// Warp NewDataset with multiple input dataset
+	filePath := path.Join(os.TempDir(), "warp.tif")
+	outputDataset, err := Warp(filePath, []*Dataset{ds1, ds2}, []string{}, CreationOption("TILED=YES"), GTiff)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer os.Remove(filePath)
+	defer outputDataset.Close()
+
+	if outputDataset.Structure().SizeX != 10 || outputDataset.Structure().SizeY != 5 {
+		t.Errorf("wrong size %d,%d", outputDataset.Structure().SizeX, outputDataset.Structure().SizeY)
+	}
+
+	data := make([]uint8, 50)
+	err = outputDataset.Read(0, 0, data, outputDataset.Structure().SizeX, outputDataset.Structure().SizeY, Bands(0, 1, 2), Window(10, 10))
+	assert.Error(t, err, "Access window out of range")
+
+	// read total warp result
+	err = outputDataset.Read(0, 0, data, outputDataset.Structure().SizeX, outputDataset.Structure().SizeY,
+		Bands(0, 1, 2),
+		Window(outputDataset.Structure().SizeX, outputDataset.Structure().SizeY),
+	)
+	assert.NoError(t, err)
+
+	assert.Equal(t, []uint8{200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 100, 100, 100, 100, 100}, data)
+}
+func TestDatasetWarpInto(t *testing.T) {
+	outputDataset, _ := Create(Memory, "", 1, Byte, 5, 5)
+	inputDataset, _ := Create(Memory, "", 1, Byte, 5, 5)
+
+	for _, b := range outputDataset.Bands() {
+		_ = b.Fill(200, 0)
+	}
+
+	for _, b := range inputDataset.Bands() {
+		_ = b.Fill(155, 0)
+	}
+
+	sr, _ := NewSpatialRefFromEPSG(4326)
+	_ = outputDataset.SetSpatialRef(sr)
+	_ = outputDataset.SetGeoTransform([6]float64{45, 1, 0, 35, 0, -1})
+	_ = inputDataset.SetSpatialRef(sr)
+	_ = inputDataset.SetGeoTransform([6]float64{45, 1, 0, 35, 0, -1})
+
+	defer outputDataset.Close()
+	defer inputDataset.Close()
+
+	// Warp existing dataset with multiple input dataset
+	err := outputDataset.WarpInto([]*Dataset{inputDataset}, []string{"-co", "TILED=YES"})
+	assert.Error(t, err, "creation option option should have error not raised")
+
+	if err = outputDataset.WarpInto([]*Dataset{inputDataset}, []string{}, ConfigOption("GDAL_CACHEMAX=64")); err != nil {
```

Comment:
the ConfigOption isn't actually tested here (i.e. you're not checking that setting a configoption results in modified behavior). I'd prefer if you leave that out as it will make it clear from the coverage report that this functionality is not actually tested. i.e.
```go
if err = outputDataset.WarpInto([]*Dataset{inputDataset},nil) ...
```